src/components/tables/EntityTable.tsx

```diff
@@ -10,51 +12,208 @@ import {
+    | TableStatuses.UNMATCHED_FILTER;
+
+type DeploymentStatus = 'ACTIVE' | 'INACTIVE';
 
 type SortDirection = 'asc' | 'desc';
 
 interface Props {
-    entities: Entity[];
-    updateDeploymentStatus: (
-        key: string,
-        deploymentStatus: DeploymentStatus
-    ) => void;
+    noExistingDataContentIds: {
+        header: string;
+        message: string;
+        docLink: string;
+        docPath: string;
+    };
+}
+
+interface Entity {
+    deploymentStatus: DeploymentStatus;
+    name: string;
+    catalogNamespace: string;
+    connectorType: string;
+    dateCreated: string;
+}
+
+interface TableState {
+    status: TableStatus;
+    error?: PostgrestError;
 }
 
 interface TableColumn {
-    field: keyof EntityMetadata | null;
+    field: keyof Entity | null;
     headerIntlKey: string;
 }
 
-function EntityTable({ entities, updateDeploymentStatus }: Props) {
+interface ConnectorInfo {
+    detail: string;
+    image_name: string;
+}
+
+const CONNECTORS_QUERY = `detail, image_name`;
+
+const LIVE_SPECS_QUERY = `spec_type, catalog_name, updated_at, connector_image_name, id`;
+
+function EntityTable({ noExistingDataContentIds }: Props) {
+    const supabaseClient: SupabaseClient = useClient();
+    const intl = useIntl();
+
+    const [tableState, setTableState] = useState<TableState>({
+        status: TableStatuses.LOADING,
+    });
+
+    const [connectors, setConnectors] = useState<any[] | undefined | null>(
+        null
+    );
+    const [publications, setPublications] = useState<any[] | undefined | null>(
+        null
+    );
+
+    const [unfilteredEntities, setUnfilteredEntities] = useState<
+        Entity[] | null
+    >(null);
+    const [filteredEntities, setFilteredEntities] = useState<Entity[] | null>(
+        null
+    );
+    const [entities, setEntities] = useState<Entity[] | null>(null);
+
     const [sortDirection, setSortDirection] = useState<SortDirection>('desc');
     const [columnToSort, setColumnToSort] =
-        useState<keyof EntityMetadata>('dateCreated');
+        useState<keyof Entity>('dateCreated');
 
     const [page, setPage] = useState(0);
     const rowsPerPage = 10;
     const rowHeight = 57;
 
-    const entityDetails: EntityMetadata[] = entities.map(
-        (entity) => entity.metadata
-    );
+    useEffect(() => {
+        (async () => {
+            const { data, error } = await supabaseClient
+                .from(TABLES.CONNECTORS)
+                .select(CONNECTORS_QUERY);
+
+            if (error) {
+                setTableState({
+                    status: TableStatuses.TECHNICAL_DIFFICULTIES,
+                    error,
+                });
+            } else {
+                setConnectors(data);
+            }
+        })().catch(() => {});
+
+        (async () => {
+            const { data, error } = await supabaseClient
+                .from(TABLES.LIVE_SPECS)
+                .select(LIVE_SPECS_QUERY)
+                .eq('spec_type', 'capture');
```

Comment:
The value `spec_type` should be evaluated against needs to be supplied via a prop for the component to remain dynamic.